### DISCUSSION TITLE
I have a small problem in Firebase Crashlytics. Please help me

### DISCUSSION THREAD

**Discussion by daniel270798** [yasirkula/UnityNativeShare#118] [Q&A] [answered]
Hi, I added your plugin to my project. I tested it on two devices (Android 7.0 and Android 9), everything works fine, but after the release in Google Play in Firebase Crashlytics I got one problem (1st screenshot). I have already read the FAQ, but I could not find a solution. Can you help me please?) On the 2nd screenshot, the script shows how I have implemented the plugin. 
P.S. English is not my native language, so there may be mistakes in the sentences...
![Screenshot_1](https://user-images.githubusercontent.com/86592403/123600290-30c85600-d7ff-11eb-844d-947651fdf5c1.png)
![Screenshot_2](https://user-images.githubusercontent.com/86592403/123600294-32921980-d7ff-11eb-803c-584532d36005.png)

**Comment by yasirkula** [comment, marked as answer]
Thank you for all the details. The issue can only come from either of these locations:

1. `getIntent().getExtras()` returning null: https://github.com/yasirkula/UnityNativeShare/blob/0a1f5020504b2abc34cbef66d7a5213be9ff0f7b/.github/AAR%20Source%20(Android)/java/com/yasirkula/unity/NativeShareCustomShareDialogActivity.java#L37
2. `getArguments()` returning null: https://github.com/yasirkula/UnityNativeShare/blob/0a1f5020504b2abc34cbef66d7a5213be9ff0f7b/.github/AAR%20Source%20(Android)/java/com/yasirkula/unity/NativeShareCustomShareDialog.java#L80

It's strange. How can these functions return null on **a fraction** of Android devices? I've googled the issue and stumbled across these topics which didn't help:

- https://stackoverflow.com/questions/38498499/android-getintent-getextras-returning-null-sometimes
- https://stackoverflow.com/questions/63035155/getextras-in-intent-is-sometimes-null
- https://stackoverflow.com/questions/30015381/getintent-getextras-is-some-times-null-in-api-level-21

Perhaps (hopefully), the top answer in this Reddit post is the reason for this behaviour: https://www.reddit.com/r/androiddev/comments/m1w338/in_activity_under_what_situation/gqfwnz0/

**Reply by daniel270798** [reply]
Thank you for such a detailed answer! I have read the materials you found. Do I understand correctly that the problem I have described has no solution and I should leave it and not take any action?)

**Reply by yasirkula** [reply]
If the problem wasn't caused by a bot as stated in the Reddit post, then I couldn't find the cause of the issue.
